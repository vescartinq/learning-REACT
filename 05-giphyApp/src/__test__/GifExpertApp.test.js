import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import { GifExpertApp } from '../GifExpertApp';

describe('GifExpertApp', () => {
    let wrapper;

    beforeEach(()=>{
        wrapper=shallow(<GifExpertApp/>)
    })

    test('should render GifExpertApp component', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should show a list of categories',()=>{
        const categories=['Avengers','Messi']
        wrapper=shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
})
