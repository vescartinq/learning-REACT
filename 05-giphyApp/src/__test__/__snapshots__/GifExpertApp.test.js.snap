// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should render GifExpertApp component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Avengers"
      key="Avengers"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Avengers"
      key="Avengers"
    />
    <GifGrid
      category="Messi"
      key="Messi"
    />
  </ol>
</Fragment>
`;
